src/messaging/messaging-types.ts

```diff
@@ -789,44 +775,49 @@ function validateLightSettings(lightSettings: LightSettings) {
       MessagingClientErrorCode.INVALID_PAYLOAD, 'android.notification.lightSettings must be a non-null object');
   }
 
-  if (typeof lightSettings.lightOnDurationMillis !== 'undefined') {
-    if (!validator.isNumber(lightSettings.lightOnDurationMillis) || lightSettings.lightOnDurationMillis < 0) {
-      throw new FirebaseMessagingError(
-        MessagingClientErrorCode.INVALID_PAYLOAD,
-        'android.notification.lightSettings.lightOnDurationMillis must be a non-negative duration in milliseconds');
-    }
-    let duration: string;
-    duration = transformMillisecondsToSecondsString(lightSettings.lightOnDurationMillis);
-    (lightSettings as any).lightOnDurationMillis = duration;
+  if (typeof lightSettings.lightOnDurationMillis === 'undefined' || lightSettings.lightOnDurationMillis === null) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD, 'android.notification.lightSettings.lightOnDurationMillis is required');
   }
-
-  if (typeof lightSettings.lightOffDurationMillis !== 'undefined') {
-    if (!validator.isNumber(lightSettings.lightOffDurationMillis) || lightSettings.lightOffDurationMillis < 0) {
-      throw new FirebaseMessagingError(
-        MessagingClientErrorCode.INVALID_PAYLOAD,
-        'android.notification.lightSettings.lightOffDurationMillis must be a non-negative duration in milliseconds');
-    }
-    const duration: string = transformMillisecondsToSecondsString(lightSettings.lightOffDurationMillis);
-    (lightSettings as any).lightOffDurationMillis = duration;
+  if (!validator.isNumber(lightSettings.lightOnDurationMillis) || lightSettings.lightOnDurationMillis < 0) {
```

Comment:
This also covers the cases where `lightOnDurationMillis` is null or undefined. Therefore you can remove the previous check.